cunyfirstapi/cunyfirstapi.py

```diff
@@ -91,12 +91,113 @@ def to_login(self):
         return self._session.get(constants.CUNY_FIRST_HOME_URL)
 
     def to_student_center(self):
-        response = self.session.get(constants.CUNY_FIRST_SIGNED_IN_STUDENT_CENTER_URL)
+        response = self._session.get(constants.CUNY_FIRST_SIGNED_IN_STUDENT_CENTER_URL)
         
         return response
 
-    def to_transcript_download(self):
-        pass
+    def to_transcript_download(self, college_code):
+        
+        # go to transcript request page without properly navigating, throw an error and get the values of the hidden form
+        url = constants.CUNY_FIRST_TRANSCRIPT_REQUEST_URL
+        response = self._session.get(url)
+        #print(r.text)
+        tree = html.fromstring(response.text)                  # parse the html DOM
+        data = {}                                       # store the key-value pairs for the form here
+        for el in tree.xpath('//input'):
+            # iterate through the hidden form
+            name = ''.join(el.xpath('./@name'))
+            value = ''.join(el.xpath('./@value'))
+            #print(name,value)
+            data[name] = value
+        
+        # manually make some changes to the form
+        data['ICAJAX'] = '1'
+        data['ICNAVTYPEDROPDOWN'] = '1'
+        data['ICYPos'] = '144'
+        data['ICStateNum'] = '1'
+        data['ICAction'] = 'DERIVED_SSS_SCR_SSS_LINK_ANCHOR4'
+        data['ICBcDomData'] = ''
+        data['DERIVED_SSS_SCL_SSS_MORE_ACADEMICS'] = '9999'
+        data['DERIVED_SSS_SCL_SSS_MORE_FINANCES'] = '9999'
+        data['CU_SF_SS_INS_WK_BUSINESS_UNIT'] = college_code
+        data['DERIVED_SSS_SCL_SSS_MORE_PROFILE'] = '9999'
+
+        # set url to student center menu
+        url = constants.CUNY_FIRST_SIGNED_IN_STUDENT_CENTER_URL
+        response = self._session.post(url = url, data = data)                 # post data to student center menu
```

Comment:
why are we navigating to Student Center here? We have a method that does that no? Why not cal the to_student_center method ?

Reply:
If the answer is because we need to pass in data then add a parameter to the `to_student_center` method that take s in data and make the data default to `None` etc

Reply:
i was thinking about that yea